tuf/repository_lib.py

```diff
@@ -92,11 +92,15 @@
 SUPPORTED_KEY_TYPES = ['rsa', 'ed25519', 'ecdsa-sha2-nistp256']
 
 
+
+
 def _generate_and_write_metadata(rolename, metadata_filename,
   targets_directory, metadata_directory, storage_backend,
   consistent_snapshot=False, filenames=None, allow_partially_signed=False,
   increment_version_number=True, repository_name='default',
-  use_existing_fileinfo=False):
+  use_existing_fileinfo=False, use_timestamp_length=True,
+  use_timestamp_hashes=True, use_snapshot_length=True,
+  use_snapshot_hashes=True):
```

Comment:
This is an internal function so it's not a show-stopper for this PR (and I don't have actionable suggestions either)... but 15 (!) arguments is a hint of something being wrong here.

Reply:
Well observed, @jku! I agree 100% that this function needs a major re-write... and also the way it is called. E.g. repository_tool's `writeall` function already performs the whole role case handling that's also done in this function, and so does `repository_lib._log_status_of_top_level_roles`, which is a whole different story (see https://github.com/theupdateframework/tuf/issues/955).

Reply:
I agree that this function is overwhelmed with arguments, but I don't think that this pr is the place to fix that.
We fixed too many bugs in this pr which are not related with `
Make length and hashes optional for timestamp and snapshot roles`.

Also, I don't see an appropriate issue that documents the need to rewrite `__generate_and_write_metadata` and how we call it.
I will create an issue about that.